packages/components/src/components/badge/badge.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import classNames from 'classnames';
+
+type TBadgeSize = 'medium' | 'large';
+type TBadgeSpacing = 'tight' | 'loose';
+type TWeight = 'bold' | 'regular';
+type TType = 'contained' | 'bordered';
+type TBackgroundColor = 'blue' | 'orange' | 'red' | 'gray';
+type TRoundedCorners = 4 | 2;
+
+interface BadgeProps extends React.HTMLAttributes<HTMLSpanElement> {
+    background_color?: TBackgroundColor;
+    className?: string;
+    color?: string;
+    rounded_corners?: TRoundedCorners;
+    size?: TBadgeSize;
+    spacing?: TBadgeSpacing;
+    type: TType;
+    weight?: TWeight;
+}
+
+const Badge = ({
+    background_color,
+    children,
+    className,
+    color,
+    weight = 'bold',
+    rounded_corners = 4,
+    size = 'medium',
+    spacing = 'tight',
+    type,
+    ...props
+}: React.PropsWithChildren<BadgeProps>) => {
+    const is_contained = type === 'contained';
+    const is_bordered = type === 'bordered';
+
+    const getBadgeStyles = () => {
+        return classNames(
+            'dc-badge',
+            {
+                'dc-badge--bold': weight === 'bold',
+                'dc-badge--regular': weight === 'regular',
+                'dc-badge--tight': spacing === 'tight',
+                'dc-badge--loose': spacing === 'loose',
+                'dc-badge__medium': size === 'medium',
+                'dc-badge__large': size === 'large',
+                'dc-badge--contained': is_contained,
+                'dc-badge--bordered': is_bordered,
+                'dc-badge--blue': is_contained && background_color === 'blue',
+                'dc-badge--orange': is_contained && background_color === 'orange',
+                'dc-badge--red': is_contained && background_color === 'red',
+                'dc-badge--gray': is_contained && background_color === 'gray',
+                'dc-badge--full-rounded': rounded_corners === 4,
+                'dc-badge--two-rounded': rounded_corners === 2,
```

Comment:
@heorhi-deriv I would suggest making two separate components the same as the design instead of handling many conditions inside one component.

<img width="784" alt="image" src="https://user-images.githubusercontent.com/72082844/235052422-c7755d83-edfa-4c78-8887-ab323c5fc048.png">

Reply:
Or if you don't want to do that then we can at least use [Discriminated Unions](https://www.totaltypescript.com/discriminated-unions-are-a-devs-best-friend) types to have better documentation. For example now if you set `type` to `bordered` you are still able to pass `background_color` but the component will ignore and don't apply the color, With discriminated unions we can make TS complain `background_color` is not a valid prop if the `type` is `bordered`.

But still I would suggest separating the components 👀

Reply:
hey @farzin-deriv thanks for review. I will apply Discriminated unions types. Regarding splitting the component into two, IMHO there are not so much styling which is related to two different types of component. most of them should be applied for both types of the badge and only few (4 colors have no relations). thanks